Software-Engineering/L5-Principal-Software-Engineer.md

```diff
@@ -1,97 +1,122 @@
+> _I am an influential Engineering leader who identifies problems and opportunities relevant to Octopus product and technical strategies, builds consensus to invest in solutions, and guides Engineering to deliver significant lasting change across Octopus._
 
-- **Planning horizon**: 1+ year
-- **Impact radius**: Group+ (15-50). Note: This role is evolving as the company grows. PSE is right at the top of our IC track. For example, the impact radius expectations of an L5 will likely grow as Octopus grows larger over the next couple years, from 2-4 teams in 2021. We expect the role to stabilise in 2022 once the company has grown to the point of needing an L6 IC engineering role.
-- **Evaluation**: Director or VP
-- **Responsibility and direction needed**: Drives technical strategy in alignment with product strategy and drives high-impact improvements across teams.
+- **Thinking horizon**: 1 year
+- **Impact radius**: Group
+- **Evaluation**: Director
+- **Responsibility and direction needed**: I drive technical strategy in alignment with product strategy for my group. I develop that technical strategy and deliver good outcomes through it, with minimal oversight.
+- **Glossary (examples from 2023/24)**: ["Octopus Technical Strategy"](https://docs.google.com/document/d/1onaZ7qRWc7BSLWyGw8duIIlOCd2k812zMIXXtlBfXL8/edit#heading=h.a65d8nqv4wgr) | ["Target Architecture"](https://github.com/OctopusDeploy/OctopusDeploy/tree/main/docs/target-architecture)
 
 ## 🦉 Domain expertise
 
-- I contribute to core technologies used by a variety of teams.
-- I help teams make critical product architecture and implementation decisions.
-- I maintain a vision of the overall technical direction within my impact radius.
-- I contribute to the engineering standards and principles of the company.
-- I contribute to the Octopus technical strategy.
-- I manage technical risks, and architect to reduce technical risk.
-- I collaborate with other Principal Software Engineers to ensure we are making cohesive and complimentary decisions that align with the Octopus technical strategy.
+- Within my group:
+    - I maintain a clear and understandable technical direction for teams within my group to help them achieve their missions with high-quality at speed.
+    - I have a deep understanding of the technical direction for all teams within my group so I can detect and correct issues early.
+    - I help teams in my group make critical product architecture and implementation decisions that result in better outcomes.
+    - I manage technical risks effectively, and architect to reduce technical risk.
+- Across Octopus:
+    - I have a broad understanding of the Octopus technical strategy and target architecture.
+    - I am an effective custodian for part of the Octopus target architecture.
```

Comment:
This seems oddly specific to me. Target architecture is important, but from my perspective it is also just another piece of important work that needs to be balanced with other PSE level work. Being able to make these tradeoffs effectively to get the right things done at the right time is the main thing I look for in a PSE rather than singling out a particular type of work. 

I think it sets a specific expectation with 5.1 PSEs that they will need to complete a target architecture project within a certain timeframe in order to be promoted to 5.2.

I think if a 5.1 PSE got to a point where they had effectively prioritised all of their work, including target architecture, and made the right tradeoff decisions to get the right things done then I would still want to promote them to 5.2. Even if at that point in time the target architecture project was unfinished or deprioritised as long as it was the right call to make.

Reply:
My interpretation of this point is that they own part of the target architecture - the thinking, analysis, and distillation into the documentation, but it does not cross over into expectations around the timely execution / delivery of it. 

PSE's should all own part of our target architecture, (I think) this is an explicit responsibility of the role - that they help us deliberately think and plan for architectural changes needed to support our business ambitions.

What if we extended it to something like:

```suggestion
    - I am an effective custodian for part of the Octopus target architecture. I contribute to the proactive architectural thinking and planning that ensures that Octopus' core technologies continue to be fit-for-purpose for the Octopus business.
```

To try and ensure that we emphasize the planning / thinking part of the contribution, but don't put expectations around execution / delivery?